### PR TITLE
MAP-181 Raid Support

### PR DESCRIPTION
**Pull request by ChrisC027**
# PR Details
## Description
The purpose of this PR is to implement the Road Authority ID (full and relative) in the message builder, the map encoder, JNI wrapper, and their associated tests.

## Related Jira Key

[MAP-181](https://usdot-carma.atlassian.net/jira/software/c/projects/MAP/board/48/backlog?selectedIssue=MAP-181)
## Motivation and Context

RoadAuthorityID is defined as Optional. However, the ASN.1 comments indicate that RAID is mandatory if roadRegulatorID is set to 0.
## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [ ] I have added tests to cover my changes.
- [] All new and existing tests passed.

[MAP-181]: https://usdot-carma.atlassian.net/browse/MAP-181?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: fedgov-cv-message-builder/src/main/java/gov/usdot/cv/msg/builder/IntersectionSituationDataBuilder.java [hunk 1]
```diff
@@ -316,10 +316,23 @@ public IntersectionGeometry[] buildIntersections(IntersectionInputData isdInputD
 			intersectionReferenceID.setRegion(referencePoint.regionID);
 		} else {
 			intersectionReferenceID.setRegionExists(false);
-			
 		}
 	
 		intersection.setId(intersectionReferenceID);
+
+		// Set full RAID or relative RAID when RRID is 0
```
Comment (dan-du-car) [inline fedgov-cv-message-builder/src/main/java/gov/usdot/cv/msg/builder/IntersectionSituationDataBuilder.java:323]: Would you update the abbreviation with full words?

File: fedgov-cv-mapencoder/src/main/java/gov/usdot/cv/mapencoder/IntersectionGeometry.java [hunk 2]
```diff
@@ -14,157 +14,201 @@
...
-    }
-}
+ package gov.usdot.cv.mapencoder;
+
+ public class IntersectionGeometry {
+     public String name;
+     private boolean nameExists;
+     private IntersectionReferenceID id;
+     private boolean fullRdAuthIDExists;
+     private String fullRdAuthID;
```
[the remark this answers is not in the document]
Comment (dan-du-car) [reply fedgov-cv-mapencoder/src/main/java/gov/usdot/cv/mapencoder/IntersectionGeometry.java:24]: https://github.com/usdot-fhwa-stol/connectedvcs-tools/pull/27

[the remark this answers is not in the document]
Comment (saina-ramyar) [reply fedgov-cv-mapencoder/src/main/java/gov/usdot/cv/mapencoder/IntersectionGeometry.java:24]: I dont think it matters here, but we need to make sure the json is read correctly. @ChrisC027 can you double check the naming?

File: fedgov-cv-message-builder/src/test/resources/samplemap8.json [hunk 3]
```diff
@@ -0,0 +1,808 @@
+{
+    "mapData": {
+      "minuteOfTheYear": 134775,
+      "layerType": "intersectionData",
+      "intersectionGeometry": {
+        "referencePoint": {
+          "descriptiveIntersctionName": "Temporary Name",
+          "intersectionID": "2580",
+          "fullRdAuthID": "1",
```
Comment (saina-ramyar) [inline fedgov-cv-message-builder/src/test/resources/samplemap8.json:9]: seems like you added "fullRdAuthID" here? Is this the correct tag?

**Review by dan-du-car — APPROVED** [review]

**Pull request merged**